### PR TITLE
Shorter Heat Equation is Hello World of CSE Blog article

### PR DESCRIPTION
**Pull request by markcmiller86**

### CODE REVIEW

File: Articles/Blog/HelloHeatEquation.md [hunk 1]
```diff
@@ -1,143 +1,99 @@
...
+Solutions to the one dimensional heat equation are also very easy for
+beginners to visualize. This can help them understand how different
+approaches work. The simplicity of the physics and mathematics also
+allows beginners to focus more on the implementation details; the
+programming language, parallel execution paradigm, discretization and
+mathematical support libraries.
+
+## A Hello CSE World Repository
+We are announcing the beginnings of an effort to populate and maintain a
+repository of implementations of the heat equation demonstrating various
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: include a link to the repository

File: Articles/Blog/HelloHeatEquation.md [hunk 2]
```diff
@@ -1,143 +1,99 @@
...
-
-While this code is almost identical the hand-coded example above, this library does include some extra features that allows users to implement some additional functionality including writing results to different kinds of R8 files, adding a timestamp, and solving for the Courant-Friedrichs-Loewy coefficient.
-
-### More Implementations!
-Using the Heat Equation to demonstrate the power of a programming model or mathematical library is not new. In fact, it is common tutorial exercise for new scientific programmers. Since there are too many interesting examples to include in this article, we've started [a repository showcase](https://github.com/betterscientificsoftware/hello-heat-equation) to collect the many examples.
+## Why this form of the Heat Equation?
+While the one dimensional heat equation might not be sufficient to
+model solar heating of the Earth, even in this relatively simple form,
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: run-on sentence?

File: Articles/Blog/HelloHeatEquation.md [hunk 3]
```diff
@@ -1,143 +1,99 @@
...
 
 ## Conclusion
-The Heat Equation is well known and has several practical applications in the fields of Mathematics and Physics. Because of this, it clearly has a wide array of implementations, from a sequential for-loop, to a portable, GPU-ready, advanced numerical library example. This exemplar PDE makes it the perfect "Hello World!" for the scientific software community. 
+The Heat Equation is well known and has many practical applications even in
+its one dimensional form. Because of this, a variety of implementations are
+easily available. This exemplar PDE makes it the perfect "Hello World!" for
+the scientific software community. Collecting a number of working examples
+together with unique implementation characteristics will help to create 
+useful introductory CSE training material. We welcome collaboration.
+To get involved, please contact one of the authors.
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: Details on getting involved can be found in the repo's contributing guide, please include a link. Remove 'contact one of the authors'.

File: Articles/Blog/HelloHeatEquation.md [hunk 4]
```diff
@@ -1,143 +1,99 @@
...
+## A Hello CSE World Repository
+We are announcing the beginnings of an effort to populate and maintain a
+repository of implementations of the heat equation demonstrating various
+aspects of CSE software. The resulting repository will be useful in training
+about fundamental aspects of the design and development of CSE software as
+well as provide examples of continuous integration testing including
+demonstrating the use of ECP-CI resources.
+
+Each implementation we gather into this repository will have to conform
+to a *minimal* set of requirements including providing a fully self-contained
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: We are not asking the examples to be runnable, only that they use common naming scheme for variables.

File: Articles/Blog/HelloHeatEquation.md [hunk 5]
```diff
@@ -1,143 +1,99 @@
...
+build, accommodating a common set of command-line arguments, producing ascii
+text files for results output and providing time and space performance data via 
+tools such as [PAPI](https://icl.utk.edu/papi/) or [Caliper](https://software.llnl.gov/Caliper/).
+
+Our goal in collecting such examples isn't so much to demonstrate the use of
+different programming languages as it is to demonstrate different approaches
+to computation and execution paradigms as well as provide instructive insights
+by comparing of time, space and scaling performance of various approaches.
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: We do not compare time, space, and scaling performance.

File: Articles/Blog/HelloHeatEquation.md [hunk 6]
```diff
@@ -1,143 +1,99 @@
...
-### Assumptions and Discretization
-To make the problem more approachable, we make some simplifying assumptions:
-* _&alpha;_ is constant
-* The heat/cooling source only comes from initial/boundary conditions
+To make the equation even more approachable, we make some simplifying assumptions:
+* _&alpha;_ is constant (the material is homogeneous)
+* Apart from initial/boundary conditions, there are no other heat sources
 * Only Cartesian coordinates in 1 dimension are used
 
-Since the heat equation takes the form of a PDE over a continuous space, most programs create a discretization of the formula over a finite set of sampling points.
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: please move the deleted text to an explanation page in the [hello-heat-equation repository](https://github.com/betterscientificsoftware/hello-heat-equation)

File: Articles/Blog/HelloHeatEquation.md [hunk 7]
```diff
@@ -1,143 +1,99 @@
...
 
-<div style="text-align:center"><img width="75%" src="https://github.com/betterscientificsoftware/images/blob/master/Blog_0719_HeatEqnBar.png" /></div>
-
-To model the temperature distribution of the rod, we can use the 1-dimensional Heat Equation given by the formula:
+## What is the Heat Equation?
+The Heat Equation, a special case of the Diffusion Equation, was developed by Joseph
+Fourier in 1822. It describes how the distribution of heat varies over space and time
+in a material. While there are many adaptations for different scenarios, beginner CSE
+programmers often start with the one-dimensional, homogeneous Heat Equation as an
+introductory problem.
```
Comment (gonsie) [inline Articles/Blog/HelloHeatEquation.md]: please add an introduction to the equation. Something like "The general heat equation is:"

File: Articles/Blog/HelloHeatEquation.md (hunk 1 above)
Comment (markcmiller86) [reply Articles/Blog/HelloHeatEquation.md]: Done. Also see [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3)

File: Articles/Blog/HelloHeatEquation.md (hunk 2 above)
Comment (markcmiller86) [reply Articles/Blog/HelloHeatEquation.md]: fixed

File: Articles/Blog/HelloHeatEquation.md (hunk 4 above)
Comment (markcmiller86) [reply Articles/Blog/HelloHeatEquation.md]: As per side-bar conversation at DOE Data Day, we agreed to keep snipits for now in the repo created for that purpose. Eventually, those will be replaced with a repo hosting fully working examples of same or similar lineage.

File: Articles/Blog/HelloHeatEquation.md (hunk 6 above)
Comment (markcmiller86) [reply Articles/Blog/HelloHeatEquation.md]: Done. See [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3)

File: Articles/Blog/HelloHeatEquation.md (hunk 5 above)
Comment (markcmiller86) [reply Articles/Blog/HelloHeatEquation.md]: As per conversation at DOE Data Day, the ultimate intention is to do so and its ok for this Blog article to announce that.

**Comment by markcmiller86** [comment]
@gonsie as per conversation, I moved text I had previously wholly deleted to the companion code snipits repository's `README.md` file. That change is in [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3). I added a line here with a link to that repo and a short explanation as to that repo's purpose. In addition, I addressed the other requests you made here.

**Comment by markcmiller86** [comment]
> Is there an official style guide for BSSW?

@gonsie this is a great question. Not to my knowledge. @rinkug?

> I would prefer to use the Oxford comma.

I had to go look up what *Oxford comma* meant. It is the comma before the *and* in a list of items as in the comma after *strawberries*.

> I like bananas, strawberries, and peaches.

FWIW, that is not my particular style preference but seeing as how you and @heatherms27 are lead authors, I will do my best to go back and correct where appropriate.

**Comment by markcmiller86** [comment]
> FWIW, that is not my particular style preference but seeing as how you and @Heatherms27 are lead authors, I will do my best to go back and correct where appropriate.

@gonsie that is done now. So, just awaiting your final review.

**Comment by rinkug** [comment]
This  is PR for issue #397

**Comment by rinkug** [comment]
We need a reviewer for this.

**Comment by rinkug** [comment]
Please refer to the next PR #425 for discussion continuation

**Pull request merged**